afs-security-analysis-local/src/main/java/com/powsybl/afs/security/local/LocalSecurityAnalysisRunningServiceExtension.java

```diff
@@ -7,39 +7,23 @@
 package com.powsybl.afs.security.local;
 
 import com.google.auto.service.AutoService;
-import com.google.common.base.Supplier;
 import com.powsybl.afs.ServiceCreationContext;
 import com.powsybl.afs.ServiceExtension;
-import com.powsybl.commons.config.ComponentDefaultConfig;
-import com.powsybl.security.SecurityAnalysisFactory;
-import com.powsybl.security.SecurityAnalysisFactoryImpl;
 import com.powsybl.afs.security.SecurityAnalysisRunningService;
 
-import java.util.Objects;
-
 /**
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
  */
 @AutoService(ServiceExtension.class)
 public class LocalSecurityAnalysisRunningServiceExtension implements ServiceExtension<SecurityAnalysisRunningService> {
 
-    private final Supplier<SecurityAnalysisFactory> factorySupplier;
```

Comment:
Here, to have the same possibility as before (choosing the security analysis implementation), we need to replace the `SecurityAnalysisFactory` by a `SecurityAnalysis.Runner`.